clients/java/src/main/java/io/camunda/zeebe/client/ZeebeClientBuilder.java

```diff
@@ -92,6 +92,19 @@
 
   ZeebeClientBuilder withJsonMapper(JsonMapper jsonMapper);
 
+  /**
+   * Overrides the authority used with TLS virtual hosting. Specifically, to override hostname
+   * verification in the TLS handshake. It does not change what host is actually connected to.
+   *
+   * <p>This method is intended for testing, but may safely be used outside of tests as an
+   * alternative to DNS overrides.
+   *
+   * <p>This setting does nothing if a {@link #usePlaintext() plaintext} connection is used.
+   *
+   * @param authority The alternative authority to use, commonly in the form <code>host:port</code>.
```

Comment:
Actually I just got this from the gRPC's own javadoc:

>Overrides the authority used with TLS and HTTP virtual hosting. It does not change what host is actually connected to. **Is commonly in the form host:port.**
If the channel builder overrides authority, any authority override from name resolution result (via EquivalentAddressGroup.ATTR_AUTHORITY_OVERRIDE) will be discarded.
This method is intended for testing, but may safely be used outside of tests as an alternative to DNS overrides.

I thought it was good to mention this. I assume that our own form will generally be just `hostname`, but that it might also contain the `port`. 

But it's good to note the origins of this. The URI javadoc actually defines authority as follows. Perhaps there's an ISO standard or RFC somewhere, but I think this is definitive enough.

> A hierarchical URI is subject to further parsing according to the syntax
>
>>    [scheme:][//authority][path][?query][#fragment] 
>
>where the characters :, /, ?, and # stand for themselves. The scheme-specific part of a hierarchical URI consists of the characters between the scheme and fragment components.
>
>The authority component of a hierarchical URI is, if specified, either server-based or registry-based. A server-based authority parses according to the familiar syntax
>
>>    [user-info@]host[:port] 
>
>where the characters @ and : stand for themselves. Nearly all URI schemes currently in use are server-based. An authority component that does not parse in this way is considered to be registry-based. 

Note that gRPC refers to this definition of authority as well: https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#forTarget-java.lang.String-

>Creates a channel with a target string, which can be either a valid [NameResolver](https://grpc.github.io/grpc-java/javadoc/io/grpc/NameResolver.html)-compliant [URI](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html?is-external=true), or an authority string.
>
>A NameResolver-compliant URI is an absolute hierarchical URI **as defined by [URI](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html?is-external=true)**.

I'll make sure to refer to the URI javadoc definition of authority.

Reply:
What do you think about something like this?

```java
/**
 * Overrides the authority used with TLS virtual hosting. Specifically, to override hostname
 * verification in the TLS handshake. It does not change what host is actually connected to.
 *
 * <p>This method is intended for testing, but may safely be used outside of tests as an
 * alternative to DNS overrides.
 *
 * <p>This setting does nothing if a {@link #usePlaintext() plaintext} connection is used.
 *
 * @param authority The alternative authority to use, commonly in the form <code>host</code> or
 *     <code>host:port</code>
 * @apiNote For the full definition of authority see [RFC 2396: Uniform Resource Identifiers
 *     (URI): Generic Syntax](http://www.ietf.org/rfc/rfc2396.txt)
 */
ZeebeClientBuilder overrideAuthority(String authority);
```

Reply:
Looks good